.github/workflows/codecov-publish.yml

```diff
@@ -0,0 +1,56 @@
+name: Codecov Publish
+
+# Due to the behavior of workflow_run, changes to this file will
+# only be reflected once it is pushed up to the `default` (main) branch
+#
+# It is recommended to test changes to this file by pushing to the default branch
+# on a fork and evaluating the correctness of the action execution
+
+on:
+  workflow_run:
+    workflows: [CI]
+    types:
+      - completed
+
+jobs:
+  publish-codecov:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download CodeCov Info
+        uses: actions/download-artifact@v4
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Can we omit all github-token references, as the action is using the default value